openshift/machine-config-operator
# OCPBUGS-43552: maxUnavailable value is not honored when disabling OCL

- What I did

Fixed a bug in the node controller where the maxUnavailable setting was not honored when ocl was disabled. The issue was due to the getAllCandidateMachines function not properly checking for nodes that were mid-update. I added a condition to skip nodes that are currently being updated, ensuring that we do not exceed the maxUnavailable limit.

- How to verify it


   1. Enable OCL in the worker pool
   2. Remove the MOSC resource to disable OCL
   3. view node status to make sure nodes do not update simultaneously

- Description for the changelog

Fixed an issue where maxUnavailable was not honored when ocl is disabled, ensuring node updates respect the maximum unavailable node count.
